### PR TITLE
Scale, canvas and textboxes

### PR DESCRIPTION
**Pull request by bytesnz**
Scale, canvas and textbox functionality is tested and stable. Only issue is the middle button issues #212  in firefox.

### CODE REVIEW

**Comment by JohnMcLear** [comment]
Great to see this stuff, I wont be able to look at this until January so hopefully someone else can peer/code review and provide suggestions!

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "ueberDB": "~0.2.1",
     "canvas": "1.2.1",
     "paper": "0.9.22",
-    "jsonminify":"~0.2.3"
+    "jsonminify":"~0.2.3",
+		"extend": "^3.0.0"
```
Comment (JohnMcLear) [inline package.json]: indentation is done by a 5 year old?

File: src/static/css/style.css [hunk 2]
```diff
@@ -333,6 +345,18 @@ input[type=checkbox] {
...
+  bottom: 10px;
+  padding: 5px;
+  color: #333;
+  background-color: rgba(255, 255, 255, 0.4);
+  border-radius: 5px;
+  border: 1px solid #ccc;
+  z-index: 1;
+}
+
 @media screen and (max-width: 600px) {
```
Comment (JohnMcLear) [inline src/static/css/style.css]: indentation

File: src/static/html/draw.html [hunk 3]
```diff
@@ -4,29 +4,29 @@
...
-	  /*
-	  |@licstart  The following is the entire license notice for the
-	  JavaScript code in this page.|
+        <script>
+          /*
+          |@licstart  The following is the entire license notice for the
+          JavaScript code in this page.|
 
-	  Copyright 2011 Peter Martischka, Primary Technology.
+          Copyright 2011 Peter Martischka, Primary Technology.
```
Comment (JohnMcLear) [inline src/static/html/draw.html]: whoops this needs changing from Peter to me as I wrote it :P

File: package.json (hunk 1 above)
Comment (theelectricwiz) [inline package.json]: Sorry for causing you such distress by the one file that I forgot to change from tabs to 2 spaces. How could I ever have been so stupid.

File: src/static/css/style.css (hunk 2 above)
Comment (theelectricwiz) [inline src/static/css/style.css]: Yes, it is indentation.

**Comment by JohnMcLear** [comment]
Hey guys, anything to stop this being merged?

**Comment by bytesnz** [comment]
Merged through other means

**Pull request closed without merging**